From b2c9384f9f62de2321484d16cd4f7c31fd34c3d3 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Mon, 27 Feb 2012 15:17:22 +0530
Subject: [PATCH 0868/1092] ASoC: McASP: Correct FIFOCTL register usage

Write FIFO register (WFIFOCTL) offset was incorrectly used in place
of Read FIFO register for McASP IP version 3. This patch corrects
the same.

TODO:
1. Differentiating the IP versions at runtime if it's possible

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 sound/soc/davinci/davinci-mcasp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/davinci/davinci-mcasp.c b/sound/soc/davinci/davinci-mcasp.c
index d2c7d7f..d6e31e9 100644
--- a/sound/soc/davinci/davinci-mcasp.c
+++ b/sound/soc/davinci/davinci-mcasp.c
@@ -401,9 +401,9 @@ static void davinci_mcasp_start(struct davinci_audio_dev *dev, int stream)
 	} else {
 		if (dev->rxnumevt) {	/* flush and enable FIFO */
 			if (dev->version == MCASP_VERSION_3) {
-				mcasp_clr_bits(dev->base + MCASP_VER3_WFIFOCTL,
+				mcasp_clr_bits(dev->base + MCASP_VER3_RFIFOCTL,
 								FIFO_ENABLE);
-				mcasp_set_bits(dev->base + MCASP_VER3_WFIFOCTL,
+				mcasp_set_bits(dev->base + MCASP_VER3_RFIFOCTL,
 								FIFO_ENABLE);
 			} else {
 				mcasp_clr_bits(dev->base + DAVINCI_MCASP_RFIFOCTL,
-- 
1.7.11.2

